bundles/org.openhab.binding.yamahamusiccast/src/main/java/org/openhab/binding/yamahamusiccast/internal/YamahaMusiccastHandler.java

```diff
@@ -0,0 +1,1244 @@
+                                    if (distributioninfo != null) {
+                                        for (JsonElement ip : distributioninfo.getClientList()) {
+                                            JsonObject clientObject = ip.getAsJsonObject();
+                                            setClientServerInfo(clientObject.get("ip_address").getAsString(), json,
+                                                    "setClientInfo");
+                                        }
+                                    }
+                                } else if ("client".equals(localRole)) {
+                                    mclinkSetupServer = connectedServer();
+                                    // Step 1. empty group on client
+                                    httpResponse = setClientServerInfo(this.host, json, "setClientInfo");
+                                    // empty zone to respect defaults
+                                    if (!"".equals(mclinkSetupServer)) {
+                                        // Step 2. remove client from server
+                                        json = "{\"group_id\":\"" + groupId
+                                                + "\", \"type\":\"remove\", \"client_list\":[\"" + this.host + "\"]}";
+                                        httpResponse = setClientServerInfo(mclinkSetupServer, json, "setServerInfo");
+                                        // Step 3. reflect changes to master
+                                        httpResponse = startDistribution(mclinkSetupServer);
+                                        localDefaultAfterMCLink = getThing().getConfiguration()
+                                                .get("defaultAfterMCLink").toString();
+                                        httpResponse = setInput(localDefaultAfterMCLink.toString(), zone, this.host);
+                                    } else if ("".equals(mclinkSetupServer)) {
+                                        // fallback in case client is removed from group by ending group on server side
+                                        localDefaultAfterMCLink = getThing().getConfiguration()
+                                                .get("defaultAfterMCLink").toString();
+                                        httpResponse = setInput(localDefaultAfterMCLink.toString(), zone, this.host);
+                                    }
+                                }
+                            }
+                        } else if ("link".equals(action)) {
+                            if (localRole != null) {
+                                if ("none".equals(localRole)) {
+                                    json = "{\"group_id\":\"" + groupId + "\", \"zone\":\"" + mclinkSetupZone
+                                            + "\", \"type\":\"add\", \"client_list\":[\"" + this.host + "\"]}";
+                                    logger.debug("setServerInfo json: {}", json);
```

Comment:
See above. Use debugger. Please check all.

Reply:
Set all _debug_ to _trace_ as this is more about verbose logging to give user insight in what is happening. At the same time, that information in necessary to solve problems.